plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduClientSession.java

```diff
@@ -613,7 +617,18 @@ private KuduSplit toKuduSplit(KuduTableHandle tableHandle, KuduScanToken token,
     {
         try {
             byte[] serializedScanToken = token.serialize();
-            return new KuduSplit(tableHandle.getSchemaTableName(), primaryKeyColumnCount, serializedScanToken, bucketNumber);
+            List<Replica> replicas = token.getTablet().getReplicas();
+            ImmutableList.Builder<HostAddress> addresses = ImmutableList.builder();
+            for (Replica replica : replicas) {
+                // KuduScanTokenBuilder uses ReplicaSelection.LEADER_ONLY by default, see org.apache.kudu.client.AbstractKuduScannerBuilder,
+                // because use ReplicaSelection.CLOSEST_REPLICA may cause slow queries when tablet followers' data lag behind tablet leaders',
+                // in this condition followers will wait until its data is synchronized with leaders' before returning
+                if (replica.getRole().toLowerCase(ENGLISH).equals("leader")) {
+                    addresses.add(fromParts(replica.getRpcHost(), replica.getRpcPort()));
+                }
+            }
```

Comment:
nit: We can also use Streams here - with the above comments.

Reply:
> nit: We can also use Streams here - with the above comments.

Thank you for your advice, i have improved the codes you mentioned above, please review again~